### PR TITLE
feat: Add Semantic Versioning

### LINKED ISSUE #16: Add semantic-release

### PR DESCRIPTION
**Pull request by devajithvs**
Fix: https://github.com/casbin/casbin-go-client/issues/16

I haven't tested it. I mainly referred https://github.com/casbin/casbin and replicated the implementation.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@nodece @sagilio @abichinger @Abingcbc plz review

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,4 +26,17 @@ jobs:
...
-      run: go test ./...
\ No newline at end of file
+      run: go test ./...
+
+  semantic-release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run semantic-release
+        # if: github.repository == 'casbin/casbin' && github.event_name == 'push'
```
Comment (sagilio) [inline .github/workflows/ci.yml]: This line is necessary to skip the run semantic-release task on PR.

**Review by sagilio — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hsluoyz) [reply .github/workflows/ci.yml]: @devajithvs

Comment (devajithvs) [reply .github/workflows/ci.yml]: Oopsie. My bad for commenting that out. I just wanted to try testing that on my repo and forgot to uncomment the line.
I'll update that

Comment (hsluoyz) [reply .github/workflows/ci.yml]: @sagilio plz review

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,4 +26,17 @@ jobs:
...
-      run: go test ./...
\ No newline at end of file
+      run: go test ./...
+
+  semantic-release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run semantic-release
+        if: github.repository == 'casbin/casbin' && github.event_name == 'push'
```
Comment (sagilio) [inline .github/workflows/ci.yml]: `github.repository == 'casbin/casbin-go-client'`

**Review by sagilio — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (nodece) [reply .github/workflows/ci.yml]: We can also create a  workflow for release this.

Comment (devajithvs) [reply .github/workflows/ci.yml]: @sagilio Done. Thanks :)
@nodece Wouldn't the action already release this? I'm sorry, I am a bit confused. Thanks

Comment (hsluoyz) [reply .github/workflows/ci.yml]: @nodece

**Review by nodece — APPROVED** [review]
LGTM

**Pull request merged**